kinoml/ml/torch_models.py

```diff
@@ -179,28 +183,182 @@ def __init__(
     ):
         super().__init__()
 
-        self.nb_char = nb_char
-        self.max_length = max_length
+        self.input_shape = input_shape
         self.embedding_shape = embedding_shape
         self.kernel_shape = kernel_shape
         self.hidden_shape = hidden_shape
         self.output_shape = output_shape
         self._activation = activation
 
         self.convolution = nn.Conv1d(
-            in_channels=self.nb_char,
+            in_channels=self.input_shape[0], # nb of characters
             out_channels=self.embedding_shape,
             kernel_size=self.kernel_shape,
         )
-        self.temp = (self.max_length - self.kernel_shape + 1) * self.embedding_shape
+        self.temp = (self.input_shape[1] - self.kernel_shape + 1) * self.embedding_shape
         self.fully_connected_1 = nn.Linear(self.temp, self.hidden_shape)
         self.fully_connected_out = nn.Linear(self.hidden_shape, self.output_shape)
 
     def forward(self, x):
         """
         Defines the foward pass for a given input 'x'
         """
+        x = x[0]
+        x = x.float()
         x = self._activation(self.convolution(x))
         x = torch.flatten(x, 1)
         x = self._activation(self.fully_connected_1(x))
         return self.fully_connected_out(x)
+
+class ConvolutionNeuralNetworkRegressionKinaseInformed(_BaseModule):
```

Comment:
Just wondering about the name: Of course KinoML is kinase-centric right now. But it was always designed to be generalizable to other protein families. Wouldn't this class also work on other protein families?

Reply:
I changed it to protein for broader use cases in the future.